src/Controllers/Index.php

```diff
@@ -1,22 +1,23 @@
 <?php
- 
-class Index {
-    
-    protected $db;
-    
-    public function __construct($config) {
-        $dbconfig = $config['database'];
-        $dsn = 'mysql:host=' . $dbconfig['host'] . ';dbname=' . $dbconfig['name'];
-        $this->db = new PDO($dsn, $dbconfig['user'], $dbconfig['pass']);
-        $this->db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
-    }
-    
+
+namespace Masterclass\Controllers;
+use Masterclass\Utils\View;
+
+/**
+ * Class Index
+ * @package Masterclass\Controllers
+ */
+class Index extends Controller{
+
+    /**
+     * @throws \Masterclass\Exceptions\BadViewDataException
```

Comment:
It comes from the view utility class
On 15 Mar 2016 07:31, "Brandon Savage" <notifications@github.com> wrote:

> In src/Controllers/Index.php
> <https://github.com/SeanJA/masterclass-repo/pull/1#discussion_r56148772>:
>
> > -        $this->db = new PDO($dsn, $dbconfig['user'], $dbconfig['pass']);
> > -        $this->db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
> > -    }
> > -
> > +
> > +namespace Masterclass\Controllers;
> > +use Masterclass\Utils\View;
> > +
> > +/**
> > + * Class Index
> > + * @package Masterclass\Controllers
> > + */
> > +class Index extends Controller{
> > +
> > +    /**
> > +     * @throws \Masterclass\Exceptions\BadViewDataException
>
> I didn't see where this gets thrown?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SeanJA/masterclass-repo/pull/1/files#r56148772
>